src/lpython/parser/parser.yy

```diff
@@ -730,6 +730,8 @@ expr
     | expr "<=" expr { $$ = COMPARE($1, LtE, $3, @$); }
     | expr ">" expr { $$ = COMPARE($1, Gt, $3, @$); }
     | expr ">=" expr { $$ = COMPARE($1, GtE, $3, @$); }
+    | expr KW_IS expr { $$ = COMPARE($1, Is, $3, @$); }
+    /* | expr KW_IS TK_NOT expr { $$ = COMPARE($1, IsNot, $4, @$); } */
```

Comment:
I had the same problem and was unable to resolve this.
cc @certik 
The problem here was that I didn't know how to assign precedence for "is not" when we uncomment the above code. We get:
 ```console
$ ./build0.sh
+ ci/version.sh
++ git describe --tags --dirty
+ version=v0.2.0-756-g1ae99e23c-dirty
+ version=0.2.0-756-g1ae99e23c-dirty
+ echo 0.2.0-756-g1ae99e23c-dirty
+ python grammar/asdl_py.py
Assuming default values of Python.asdl and python_ast.py
+ python grammar/asdl_cpp.py grammar/Python.asdl src/lpython/python_ast.h
+ python grammar/asdl_cpp.py src/libasr/ASR.asdl src/libasr/asr.h
+ python src/libasr/wasm_instructions_visitor.py
Assuming default values of wasm_instructions.txt and wasm_visitor.h
+ cd src/lpython/parser
+ re2c -W -b tokenizer.re -o tokenizer.cpp
+ cd src/lpython/parser
+ bison -Wall -d -r all parser.yy
parser.yy: error: reduce/reduce conflicts: 27 found, 0 expected

```

Reply:
That works, Thank you!
```diff
diff --git a/src/lpython/parser/parser.yy b/src/lpython/parser/parser.yy
index f3a7d7797..0954e071b 100644
--- a/src/lpython/parser/parser.yy
+++ b/src/lpython/parser/parser.yy
@@ -131,6 +131,7 @@ void yyerror(YYLTYPE *yyloc, LFortran::Parser &p, const std::string &msg)
 %token TK_GT ">"
 %token TK_GE ">="
 %token TK_NOT "not"
+%token TK_IS_NOT "is not"
 %token TK_AND "and"
 %token TK_OR "or"
 %token TK_TRUE "True"
@@ -243,7 +244,7 @@ void yyerror(YYLTYPE *yyloc, LFortran::Parser &p, const std::string &msg)
 %left "or"
 %left "and"
 %precedence "not"
-%left "==" "!=" ">=" ">" "<=" "<" KW_IS //"is not" "is" "not in" "in"
+%left "==" "!=" ">=" ">" "<=" "<" KW_IS "is not" // "is" "not in" "in"
 %left "|"
 %left "^"
 %left "&"
@@ -731,7 +732,7 @@ expr
     | expr ">" expr { $$ = COMPARE($1, Gt, $3, @$); }
     | expr ">=" expr { $$ = COMPARE($1, GtE, $3, @$); }
     | expr KW_IS expr { $$ = COMPARE($1, Is, $3, @$); }
-    /* | expr KW_IS TK_NOT expr { $$ = COMPARE($1, IsNot, $4, @$); } */
+    | expr TK_IS_NOT expr { $$ = COMPARE($1, IsNot, $3, @$); }
 
     | expr "and" expr { $$ = BOOLOP($1, And, $3, @$); }
     | expr "or" expr { $$ = BOOLOP($1, Or, $3, @$); }
diff --git a/src/lpython/parser/tokenizer.re b/src/lpython/parser/tokenizer.re
index f2edf0d2d..032df4462 100644
--- a/src/lpython/parser/tokenizer.re
+++ b/src/lpython/parser/tokenizer.re
@@ -426,6 +426,7 @@ int Tokenizer::lex(Allocator &al, YYSTYPE &yylval, Location &loc, diag::Diagnost
             'not'  { RET(TK_NOT) }
             'and'  { RET(TK_AND) }
             'or'   { RET(TK_OR) }
+            'is not'  { RET(TK_IS_NOT) }
```
But, more than one whitespace between `is` and `not` does not work like
```python
if type(x) is  not int:
    pass
```

Reply:
@akshanshbhatt Apply the above diff. This should work

Reply:
> I think we might need to parse `is not` as a token in the tokenizer.

This is exactly was I was going to suggest. Will implement it ASAP!